### ISSUE TITLE
Doesn't do anything after Hooks completed

### ISSUE THREAD

**Issue by angelux079** [wvdumper/dumper#21]
frida-server runs fine and when the script runs and says Hooks completed, I play widevine DRM content and it doesn't give me anything else. I used bitmovin.com/demos/drm to play DRM content

**Comment by zackmark29** [comment]
Did you check if Widevine is enabled on your browser?
![image](https://user-images.githubusercontent.com/62680932/147193582-7e319b0e-f425-401b-9ba0-2f684dd97a31.png)
![image](https://user-images.githubusercontent.com/62680932/147193624-a16d6995-8368-4b1d-a88b-65b9e3faa917.png)

**Comment by angelux079** [comment]
> Did you check if Widevine is enabled in your browser? ![image](https://user-images.githubusercontent.com/62680932/147193582-7e319b0e-f425-401b-9ba0-2f684dd97a31.png) ![image](https://user-images.githubusercontent.com/62680932/147193624-a16d6995-8368-4b1d-a88b-65b9e3faa917.png)

yes, drm content played back fine with widevine but keys did not dump

**Comment by flex36ty** [comment]
What version of android / magisk are you running?

**Comment by angelux079** [comment]
> What version of android / magisk are you running?

android 5.1, my device is a galaxy core prime from t mobile (SM-G360T)
frida-server runs fine and script too but it just doesn't do anything after Hooks completed

**Comment by flex36ty** [comment]
> > What version of android / magisk are you running?
> 
> android 5.1, my device is a galaxy core prime from t mobile (SM-G360T) frida-server runs fine and script too but it just doesn't do anything after Hooks completed

Do you have liboemcrytodosabler installed too?

**Comment by angelux079** [comment]
> > > What version of android / magisk are you running?
> > 
> > 
> > android 5.1, my device is a galaxy core prime from t mobile (SM-G360T) frida-server runs fine and script too but it just doesn't do anything after Hooks completed
> 
> Do you have liboemcrytodosabler installed too?

I don't have magisk since I wasn't able to install it but my phone is rooted

**Comment by JD906** [comment]
FYI, I don't think Samsung is good for this.
I've tested several models,, no luck.
But OnePlus phone succeeded.

**Comment by zackmark29** [comment]
> > What version of android / magisk are you running?
> 
> android 5.1, my device is a galaxy core prime from t mobile (SM-G360T) frida-server runs fine and script too but it just doesn't do anything after Hooks completed

Ok. Here's the hint. Upgrade it to at least version 7. You can use lineageOS.
Mine also Samsung Tablet 5.1 I only succeed after upgrading to version 7. But still there is more to do/troubleshoot after upgrading.

**Comment by projectnaked** [comment]
galaxy note 3 worked fine for me

**Comment by projectnaked** [comment]
> > > > What version of android / magisk are you running?
> > > 
> > > 
> > > android 5.1, my device is a galaxy core prime from t mobile (SM-G360T) frida-server runs fine and script too but it just doesn't do anything after Hooks completed
> > 
> > 
> > Do you have liboemcrytodosabler installed too?
> 
> I don't have magisk since I wasn't able to install it but my phone is rooted

which version of magisk did you tried?

**Comment by JD906** [comment]
> galaxy note 3 worked fine for me

Ok, good to know.
What cdm system id did you get?

**Comment by angelux079** [comment]
what frida server can run on android 5? I tried some versions, but they just crash the phone and I'm using the correct architecture

**Comment by Leesnow0204** [comment]
I run the frida on  andriod , but I don't know how to continue ?

can you help me? 
![未命名](https://user-images.githubusercontent.com/96590223/147312350-d5f693fa-adaa-4a55-ba38-ce9451c0966b.jpg)

**Comment by angelux079** [comment]
I think it stays on Hooks completed because OEM Crypto API is missing, you can see with DRM Info

**Comment by Rus4kova** [comment]
> 

Worked fine on my Samsung Galaxy S7 Edge with android 8.0

**Comment by Rus4kova** [comment]
> FYI, I don't think Samsung is good for this.
> I've tested several models,, no luck.
> But OnePlus phone succeeded.

Worked fine on my Samsung Galaxy S7 Edge with android 8.0

**Comment by Rus4kova** [comment]
For anyone trying to dump from Samsung phones ...
1. Install samsung USB drivers
2. Install adb from google - just use the ZIP file no need to install
3. before running script - start adb server and type **adb devices** in console to make sure your phone is connected and visible
4. run the script and start the netflix app.

I have dumped L3's from 2 x S7 Edge and S9+ phones. No issues all using Magisk and installing Frida through Magisk so it starts up when the phone boots.

**Comment by evildemons** [comment]
> For anyone trying to dump from Samsung phones ...
> 
> 1. Install samsung USB drivers
> 2. Install adb from google - just use the ZIP file no need to install
> 3. before running script - start adb server and type **adb devices** in console to make sure your phone is connected and visible
> 4. run the script and start the netflix app.
> 
> I have dumped L3's from 2 x S7 Edge and S9+ phones. No issues all using Magisk and installing Frida through Magisk so it starts up when the phone boots.

I have attempts that you have written without success. What am I doing wrong do you have discord?

**Comment by Rus4kova** [comment]
> > For anyone trying to dump from Samsung phones ...
> > 
> > 1. Install samsung USB drivers
> > 2. Install adb from google - just use the ZIP file no need to install
> > 3. before running script - start adb server and type **adb devices** in console to make sure your phone is connected and visible
> > 4. run the script and start the netflix app.
> > 
> > I have dumped L3's from 2 x S7 Edge and S9+ phones. No issues all using Magisk and installing Frida through Magisk so it starts up when the phone boots.
> 
> I have attempts that you have written without success. What am I doing wrong do you have discord?

It's impossible to say what went wrong, when you don't give any information about errors messages etc.

**Comment by evildemons** [comment]
> It's impossible to say what went wrong, when you don't give any information about errors messages etc.

I only get keybox and no key. Which os version have you had on your samsung?

**Comment by 89z** [comment]
Same problem  here -  @angelux079 - what is that site in the screenshot?

**Comment by larsk2** [comment]
Don't use Netflix. Use a free "DRM test" site for obtaining keys.

**Comment by 89z** [comment]
I am trying to get this to work with Android Studio, is it possible?

**Comment by projectnaked** [comment]
> I am trying to get this to work with Android Studio, is it possible?

yes it is

**Comment by 89z** [comment]
I finally figured it out - the problem is the  wvdumper output is confusing - you get an error - but if you just click LOAD after Hooks completed - then you get the keys. you don't need Magisk or anything

**Comment by grazeloon** [comment]
> I finally figured it out - the problem is the wvdumper output is confusing - you get an error - but if you just click LOAD after Hooks completed - then you get the keys. you don't need Magisk or anything

what does LOAD mean, how do I click LOAD on terminal ?

**Comment by 89z** [comment]
https://bitmovin.com/demos/drm

**Comment by vathanac** [comment]
> I finally figured it out - the problem is the wvdumper output is confusing - you get an error - but if you just click LOAD after Hooks completed - then you get the keys. you don't need Magisk or anything

I rooted Noxplayer with [magiskdelta ](https://huskydg.github.io/magisk-files/#how-to-install-magisk-into-emulator-such-as-noxplayer-ldplayer-etc). Connect emulator to pc via debug bridge. Run dump_keys.py and play video on emulator([it said No DRm in chrome](https://i.imgur.com/IY0wSXC.png)). I tried again and again, still no luck :(


I don't know what to do anymore.

**Comment by 89z** [comment]
@vathanac sorry I don't use any Magisk bullshit. I only use Android Studio or GenyMotion

**Comment by jerrydev010** [comment]
> > It's impossible to say what went wrong, when you don't give any information about errors messages etc.
> 
> I only get keybox and no key. Which os version have you had on your samsung?

@evildemons Email me, I might have something for you 
PS. This is free 

kekw010@protonmail.com

**Comment by Abdelraman** [comment]
any luck with genymotion, android 8 google pixel 3
mentioning hooks completed and stuck at that

**Comment by 3052** [comment]
I finally figured it out - the problem is the wvdumper output is confusing - you get an error - but if you just click LOAD after Hooks completed - then you get the keys. you don't need Magisk or anything

**Comment by Abdelraman** [comment]
@3052 exactly, but opening website like udemy in a webview stucks since the web view is not updated

**Comment by 3052** [comment]
you dont need to use Udemy at all. the point of this tool is to dump a CDM. once you have done that, you can use the same CDM with many different websites. thats why you typically see basic websites like this being recommended:

https://integration.widevine.com/player

because you just need the simplest website that makes a license request. ideally the site you use to dump will be public and wont even require a login, like the one I just linked.
